app/uk/gov/hmrc/cataloguefrontend/whatsrunningwhere/WhatsRunningWhereController.scala

```diff
@@ -67,41 +69,55 @@ class WhatsRunningWhereController @Inject()(
   def ecs: Action[AnyContent] =
     Action.async { implicit request =>
       for {
-        form                 <- Future.successful(WhatsRunningWhereFilter.form.bindFromRequest)
-        profile              =  profileFrom(form)
-        selectedProfileType  =  form.fold(_ => None, _.profileType).getOrElse(ProfileType.Team)
+        form <- Future.successful(WhatsRunningWhereFilter.form.bindFromRequest)
+        profile             = profileFrom(form)
+        selectedProfileType = form.fold(_ => None, _.profileType).getOrElse(ProfileType.Team)
         (releases, profiles) <- (service.ecsReleases(profile), service.profiles).mapN { case (r, p) => (r, p) }
-        environments         =  distinctEnvironments(releases)
-        profileNames         =  profiles.filter(_.profileType == selectedProfileType).map(_.profileName).sorted
-      } yield
-        Ok(page(environments, releases, selectedProfileType, profileNames, form))
+        environments = distinctEnvironments(releases)
+        profileNames = profiles.filter(_.profileType == selectedProfileType).map(_.profileName).sorted
+      } yield Ok(page(environments, releases, selectedProfileType, profileNames, form))
+    }
+
+  def ecsAndHeritage: Action[AnyContent] =
+    Action.async { implicit request =>
+      for {
+        form <- Future.successful(WhatsRunningWhereFilter.form.bindFromRequest)
+        profile             = profileFrom(form)
+        selectedProfileType = form.fold(_ => None, _.profileType).getOrElse(ProfileType.Team)
+        (ecsReleases, profiles) <- (service.ecsReleases(profile), service.profiles).mapN { case (r, p) => (r, p) }
+        heritageReleases        <- service.releases(profile)
+        releases     = (ecsReleases ++ heritageReleases).sortBy(_.applicationName.asString)
+        environments = distinctEnvironments(releases)
+        profileNames = profiles.filter(_.profileType == selectedProfileType).map(_.profileName).sorted
+      } yield Ok(page(environments, releases, selectedProfileType, profileNames, form))
     }
 }
```

Comment:
Is it worth refactoring since `heritage`, `ecs` and `ecsAndHeritage` are so similar, except the additional `service.releases(profile)`?
E.g. heritage = ecsAndHeritage(ecs = false, heritage = true)
with
`if (ecs) service.ecsReleases else Future(Seq.empty)` etc.

Reply:
makes sense ecsAndHeritage isn't used (as other colin commented) seems like a good way of removing it but retaining functionality for the future